services/api-server/src/simcore_service_api_server/api/routes/solvers_jobs.py

```diff
@@ -273,49 +230,42 @@ async def get_job_outputs(
-
-    async def _draft_impl():
-        job_name = compose_resource_name(solver_key, version, job_id)
-        logger.debug("Get Job '%s' outputs", job_name)
-
-        project: Project = await webserver_api.get_project(project_id=job_id)
-        node_ids = list(project.workbench.keys())
-        assert len(node_ids) == 1  # nosec
-
-        outputs: Dict[
-            str, Union[float, int, bool, BaseFileLink, str, None]
-        ] = await get_solver_output_results(
-            user_id=user_id,
-            project_uuid=job_id,
-            node_uuid=UUID(node_ids[0]),
-            db_engine=db_engine,
-        )
-
-        results: Dict[str, ArgumentType] = {}
-        for name, value in outputs.items():
-            if isinstance(value, BaseFileLink):
-                # TODO: value.path exists??
-                file_id: UUID = File.create_id(*value.path.split("/"))
-
-                # TODO: acquire_soft_link will halve calls
-                found = await storage_client.search_files(user_id, file_id)
-                if found:
-                    assert len(found) == 1
-                    results[name] = to_file_api_model(found[0])
-                else:
-                    api_file: File = await storage_client.create_soft_link(
-                        user_id, value.path, file_id
-                    )
-                    results[name] = api_file
-            else:
-                # TODO: cast against catalog's output port specs
-                results[name] = value
 
-        job_outputs = JobOutputs(job_id=job_id, results=results)
-        return job_outputs
-
-    return await _draft_impl()
+    job_name = _compose_job_resource_name(solver_key, version, job_id)
+    logger.debug("Get Job '%s' outputs", job_name)
+
+    project: Project = await webserver_api.get_project(project_id=job_id)
+    node_ids = list(project.workbench.keys())
+    assert len(node_ids) == 1  # nosec
+
+    outputs: Dict[
+        str, Union[float, int, bool, BaseFileLink, str, None]
+    ] = await get_solver_output_results(
+        user_id=user_id,
+        project_uuid=job_id,
+        node_uuid=UUID(node_ids[0]),
+        db_engine=db_engine,
+    )
+
+    results: Dict[str, ArgumentType] = {}
+    for name, value in outputs.items():
+        if isinstance(value, BaseFileLink):
+            # TODO: value.path exists??
+            file_id: UUID = File.create_id(*value.path.split("/"))
+
+            # TODO: acquire_soft_link will halve calls
+            found = await storage_client.search_files(user_id, file_id)
+            if found:
+                assert len(found) == 1
+                results[name] = to_file_api_model(found[0])
+            else:
+                api_file: File = await storage_client.create_soft_link(
+                    user_id, value.path, file_id
+                )
+                results[name] = api_file
+        else:
+            # TODO: cast against catalog's output port specs
```

Comment:
is this still relevant?

Reply:
I need to do another iteration on error handling. It is a reminder for next PR.